bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/api/WindcentraleAPI.java

```diff
@@ -0,0 +1,148 @@
+import org.openhab.binding.windcentrale.internal.exception.InvalidAccessTokenException;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WindcentraleAPI} implements the Windcentrale REST API which allows for querying project participations and
+ * the current windmill status.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@NonNullByDefault
+public class WindcentraleAPI {
+
+    public static final String URL_PREFIX = "https://mijn.windcentrale.nl/api/v0";
+    private static final String LIVE_DATA_URL = URL_PREFIX + "/livedata";
+    private static final String PROJECTS_URL = URL_PREFIX + "/sustainable/projects";
+
+    private static final String APPLICATION_JSON = "application/json";
+    private static final String BEARER = "Bearer ";
+    private static final Duration REQUEST_TIMEOUT = Duration.ofMinutes(1);
+
+    private final Logger logger = LoggerFactory.getLogger(WindcentraleAPI.class);
+
+    private final HttpClient httpClient;
+    private final TokenProvider tokenProvider;
+
+    private final Set<RequestListener> requestListeners = ConcurrentHashMap.newKeySet();
+
+    public WindcentraleAPI(HttpClientFactory httpClientFactory, TokenProvider tokenProvider) {
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+        this.tokenProvider = tokenProvider;
+    }
+
+    public void dispose() {
+        requestListeners.clear();
+    }
+
+    public void addRequestListener(RequestListener listener) {
+        requestListeners.add(listener);
+    }
+
+    public void removeRequestListener(RequestListener listener) {
+        requestListeners.remove(listener);
+    }
+
+    private String getAuthorizationHeader() throws InvalidAccessTokenException {
+        return BEARER + tokenProvider.getIdToken();
+    }
+
+    private String getJson(String url) throws FailedGettingDataException, InvalidAccessTokenException {
+        try {
+            logger.debug("Getting JSON from: {}", url);
+            ContentResponse contentResponse = httpClient.newRequest(url) //
+                    .method(GET) //
+                    .header(ACCEPT, APPLICATION_JSON) //
+                    .header(AUTHORIZATION, getAuthorizationHeader()) //
+                    .timeout(REQUEST_TIMEOUT.toNanos(), TimeUnit.NANOSECONDS) //
+                    .send();
+
+            if (contentResponse.getStatus() >= 400) {
+                FailedGettingDataException exception = new FailedGettingDataException(
+                        String.format("Windcentrale API error: %s (HTTP %s)", contentResponse.getReason(),
+                                contentResponse.getStatus()));
+                requestListeners.forEach(listener -> listener.onError(exception));
+                throw exception;
+            }
+            String response = contentResponse.getContentAsString();
+            logger.debug("Response: {}", response);
```

Comment:
Since full response payload is logged, probably:
```suggestion
            logger.trace("Response: {}", response);
```